### PR TITLE
Enable SAI_SWITCH_ATTR_UNINIT_DATA_PLANE_ON_REMOVAL attribute

### PR DESCRIPTION
**Pull request by tbgowda**
Enable SAI_SWITCH_ATTR_UNINIT_DATA_PLANE_ON_REMOVAL attribute for all platforms based on capability

- Remove Mellanox specific checks
- Check whether the attribute is supported
- Call SAI API if it is supported, else ignore

Cherry-picking https://github.com/Azure/sonic-sairedis/pull/975

### CODE REVIEW

**Comment by tbgowda** [comment]
This is a double commit of https://github.com/Azure/sonic-sairedis/pull/975.

a. it is tested on cisco platforms
b. don't have access to other platforms

File: syncd/Syncd.cpp [hunk 1]
```diff
@@ -4181,15 +4181,36 @@ sai_status_t Syncd::setUninitDataPlaneOnRemovalOnAllSwitches()
 
         auto strRid = sai_serialize_object_id(rid);
 
-        auto status = m_vendorSai->set(SAI_OBJECT_TYPE_SWITCH, rid, &attr);
+        sai_attr_capability_t attr_capability = {};
 
-        if (status != SAI_STATUS_SUCCESS)
+        sai_status_t queryStatus;
+
+        queryStatus = sai_query_attribute_capability(rid,
```
Comment (kcudnik) [inline syncd/Syncd.cpp:4188]: don't use SAI api directly, use m_vendorSai

Comment (tbgowda) [reply syncd/Syncd.cpp]: Apologies for the delay. I have addressed this comment. Please take a look.

**Comment by tbgowda** [comment]
Thank you! will do. Please help merge the change so that it can be released into sonic-buildimage.

**Pull request merged**